docker/Dockerfile.exp

```diff
@@ -2,18 +2,19 @@
 # docker build -t esmvaltool:experimental . -f docker/Dockerfile.exp
 FROM esmvalgroup/esmvalcore:latest
 
+SHELL ["/bin/bash", "--login", "-c"]
+
 COPY ./environment.yml /src/ESMValTool/environment.yml
 WORKDIR /src/ESMValTool
-RUN conda update -n base -y conda && conda install -y -c conda-forge mamba && mamba env create --name esmvaltool --file environment.yml && conda clean --all -y
+RUN conda update -n base -y conda && conda install -y -n base -c conda-forge mamba && mamba env update -n esmvaltool -f environment.yml && conda clean --all -y
 
-# Make RUN commands use the new environment:
 SHELL ["conda", "run", "--name", "esmvaltool", "/bin/bash", "-c"]
 
 COPY . /src/ESMValTool
+RUN conda info && conda list
 RUN pip install -e .[develop]
-RUN mamba install -y -c conda-forge r-base
+RUN conda info && conda list
 RUN esmvaltool install R
-RUN mamba install -y -c conda-forge julia
```

Comment:
you are correct, Klaus, take back that what I said, I actually checked the previous (succesful, before you simplified) run and r-base was installed only with minimal changes in deps ie the package was there before calling the install and with the correct version, and julia was perfectly installed too, mamba saying "nothing to do" - so your simplification is valid, and also, now it's crappe-ing because yet another issue with Julia it seems

Reply:
Julia...

Reply:
OK I confirm I can reproduce that locally on my laptop but only when building the container, if installing Julia via `esmvaltool install Julia` normally (after a fresh env creation, esmvaltool installation) all works fine

Reply:
OK this is getting weired and weirder, this is the partial tail of the error message (it's humongous, it's literally conda trying to establish which package is in conflict with which):
```
julia -> openlibm[version='<0.8.0']The following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.28=0
  - feature:|@/linux-64::__glibc==2.28=0
  - arpack -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - atk-1.0 -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - blosc -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - brotlipy==0.7.0=py39h27cfd23_1003 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - bzip2 -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - cffi==1.15.0=py39hd667e15_1 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - conda-package-handling==1.7.3=py39h27cfd23_1 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - cryptography==36.0.0=py39h9ce1e76_0 -> libgcc-ng -> __glibc[version='>=2.17']
  - cytoolz -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - fiona -> libgdal[version='>=3.4.1,<3.5.0a0'] -> __glibc[version='>=2.17,<3.0.a0']
  - freetype -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - freexl -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - fribidi -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - gdal -> libgdal==3.4.1=haddd6ec_5 -> __glibc[version='>=2.17,<3.0.a0']
  - gdk-pixbuf -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - gfortran_impl_linux-64 -> libgcc-ng[version='>=4.9'] -> __glibc[version='>=2.17']
  - giflib -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - gmp -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - graphite2 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - gsl -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - gtk2 -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - gts -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - hdf4 -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - hdfeos2 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - jbig -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - json-c -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - lcms2 -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - libdap4 -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - libdeflate -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - libedit -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - libev -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - libffi==3.3=he6710b0_2 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - libgcc-ng==9.3.0=h5101ec6_17 -> __glibc[version='>=2.17']
  - libgdal -> __glibc[version='>=2.17,<3.0.a0']
  - libglu -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - libiconv -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - libkml -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - libpng -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - libstdcxx-ng==9.3.0=hd4cf53a_17 -> __glibc[version='>=2.17']
  - libunwind -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - libuuid -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - libxcb -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - libzlib -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - lz4-c -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - make -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - markupsafe -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - metis -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - mpfr -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - mpich -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - msgpack-python -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - ncl -> libgdal[version='>=3.4.0,<3.5.0a0'] -> __glibc[version='>=2.17,<3.0.a0']
  - ncurses==6.3=h7f8727e_2 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - openjpeg -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - openssl==1.1.1m=h7f8727e_0 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - ossuuid -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - pcre -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - pcre2 -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - pixman -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - pkg-config -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - psutil -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - pthread-stubs -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - pycosat==0.6.3=py39h27cfd23_0 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - python-stratify -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - python-xxhash -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - python==3.9.7=h12debd9_1 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - r-askpass -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - r-curl -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - r-ellipsis -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - r-lazyeval -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - r-processx -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - r-ps -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - r-purrr -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - r-sys -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - r-utf8 -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - r-vctrs -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - rasterio -> libgdal[version='>=3.4.1,<3.5.0a0'] -> __glibc[version='>=2.17,<3.0.a0']
  - readline==8.1.2=h7f8727e_1 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - ruamel_yaml==0.15.100=py39h27cfd23_0 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - sed -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - sqlite==3.37.0=hc218d9a_0 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - tk==8.6.11=h1ccaba5_0 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - tktable -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17']
  - tornado -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-fixesproto -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-inputproto -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-kbproto -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libice -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libsm -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libx11 -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libxau -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libxdmcp -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libxext -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libxfixes -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libxi -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libxmu -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libxpm -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libxrender -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-libxt -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-makedepend -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - xorg-renderproto -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-xextproto -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xorg-xproto -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xxhash -> libgcc-ng[version='>=9.3.0'] -> __glibc[version='>=2.17']
  - xz==5.2.5=h7b6447c_0 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - yaml==0.2.5=h7b6447c_0 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
  - zlib==1.2.11=h7f8727e_4 -> libgcc-ng[version='>=7.5.0'] -> __glibc[version='>=2.17'  -   -   -   -   -   -   -   -   -   -   -   -   -   
  
  Your installed version is: 2.28
```